pkg/api/v1alpha1/cluster.go

```diff
@@ -591,12 +591,14 @@ func validateProxyData(proxy string) error {
 	} else {
 		proxyHost = proxy
 	}
-	ip, port, err := net.SplitHostPort(proxyHost)
+	host, port, err := net.SplitHostPort(proxyHost)
 	if err != nil {
 		return fmt.Errorf("proxy %s is invalid, please provide a valid proxy in the format proxy_ip:port", proxy)
 	}
-	if net.ParseIP(ip) == nil {
-		return fmt.Errorf("proxy ip %s is invalid, please provide a valid proxy ip", ip)
+	if net.ParseIP(host) == nil {
+		if _, err := net.LookupIP(host); err != nil {
+			return fmt.Errorf("proxy ip %s is invalid, please provide a valid proxy ip", host)
```

Comment:
how about something like?
```suggestion
			return fmt.Errorf("proxy endpoint %s is invalid, please provide a valid proxy domain name or ip", host)
```